src/magellan_vision/scripts/line_detector.py

```diff
@@ -1,211 +1,159 @@
-                    cv2.line(crop2, (x1, y1), (x2, y2), (0, 255, 0), 3)
-                    # cv2.circle(crop, (x1, y1),(5), (0,255,0), 3)
-                    # cv2.circle(crop, (x2, y2),(5), (0,255,0), 3)
-            circles = crop
-            lines = crop2
-
-    if linesP is None or len(linesP) < 40:
-        skipped_frames += 1
-
-    cv2.imshow('Crop', crop)
-    # linesAndCircles = cv2.addWeighted(lines, 0.5, circles, 0.5,0)
-    # final = cv2.addWeighted(processed, 0.8, linesAndCircles, 0.5,0)
-    # cv2.imshow('Original', frame)
-    # cv2.imshow('Combo', final)
-    # print(skipped_frames)
-
-    # FIXME: what is this crap, please FIX ME
-    # plt.plot(np.unique(x_array_right),
-    #          np.poly1d(np.polyfit(x_array_right, y_array_right, 1))
-    #          (np.unique(x_array_right)))
-
-    # right_line_slope,right_line_intercept = best_fit(np.asarray(x_array_right),np.asarray(y_array_right))
-    # left_line_slope,left_line_intercept = best_fit(np.asarray(x_array_left),np.asarray(y_array_left))
-
-    toc = time.time()
-    delta_t = toc - tic
-    print("FPS: {}".format(1/delta_t))
-
-    key = cv2.waitKey(25)
-    if key == 27:
-        break
-
-video.release()
-cv2.destroyAllWindows
+
+# Numpy and OpenCV
+import cv2
+import numpy as np
+
+# ROS imports
+import rospy
+from sensor_msgs.msg import Image  # ROS Image Message
+from cv_bridge import CvBridge, CvBridgeError  # Converts b/w OpenCV Image and ROS Image Message
+from magellan_core.msg import Float64Arr  # ROS Float64Arr type msg
+
+# Global Constant
+VERBOSE = True
+
+
+# Class PubSubNode: Subscribes to the /camera/color/image_raw topic(RGB Images from realsense camera) and publishes
+#                   to the /perception/color/image_processed topic(Processed RGB raw Images) and to the
+#                   /perception/detectedPoints topic(Detected Points as a Float64Arr msg)
+# TODO: Publish to other topics: Depth image; etc..
+class PubSubNode:
+    def __init__(self):
+        '''Initialize ros publisher, ros subscriber'''
+        self._lines_object = Lines()
+        # topic where we publish
+        self._image_pub = rospy.Publisher("/perception/color/image_processed", Image, queue_size=5)
+        self._point_arr_pub = rospy.Publisher('/perception/detectedPoints', Float64Arr, queue_size=10)
+        self._bridge = CvBridge()
+
+        # subscribed Topic
+        self._subscriber = rospy.Subscriber("/camera/color/image_raw", Image, self._callback, queue_size=1)
+        if VERBOSE:
+            print "subscribed to /camera/color/image_raw"
+
+    def _callback(self, ros_data):
+        '''Callback function of subscribed topic.
+        Here images get converted and features detected and published'''
+        if VERBOSE:
+            print 'received image of type: "%s"' % type(ros_data)
+        time0 = time.time()
+        # conversion to cv2
+        try:
+            image_np = self._bridge.imgmsg_to_cv2(ros_data, "bgr8")
+        except CvBridgeError:
```

Comment:
It might be worth logging if this fails

Reply:
@camkisailus how was this resolved, I don't see the log

Reply:
@camkisailus this should not be a debug message, it should be an error